lib/openhab/core_ext/java/zoned_date_time.rb

```diff
@@ -90,6 +90,117 @@ def to_zoned_date_time(context = nil) # rubocop:disable Lint/UnusedMethodArgumen
+        # @return [Symbol, nil]
+        #
+        # @example
+        #   MonthDay.parse("12-25").holiday # => :christmas
+        #
+        def holiday(file: nil)
+          (if file
+             ::Ephemeris.get_bank_holiday_name(self,
+                                               file)
+           else
+             ::Ephemeris.get_bank_holiday_name(self)
+           end)&.downcase&.to_sym
+        end
+
+        #
+        # Description of the holiday for this date.
+        #
+        # @param [String, nil] file Optional file to use for holiday definitions.
+        # @return [String, nil]
+        #
+        # @example
+        #   MonthDay.parse("12-25").holiday_description # => "Christmas"
+        #
+        def holiday_description(file: nil)
+          holiday(file: file)&.then { |h| ::Ephemeris.get_holiday_description(h.to_s.upcase) }
+        end
+
+        #
+        # Determines if this date is on a holiday.
+        #
+        # @param [String, nil] file Optional file to use for holiday definitions.
+        # @return [true, false]
+        #
+        def holiday?(file: nil)
+          file ? ::Ephemeris.bank_holiday?(self, file) : ::Ephemeris.bank_holiday?(self)
+        end
+
+        #
+        # Name of the closest holiday on or after this date.
+        #
+        # @param [String, nil] file Optional file to use for holiday definitions.
+        # @return [Symbol]
+        #
+        def next_holiday(file: nil)
+          (file ? ::Ephemeris.get_next_bank_holiday(self, file) : ::Ephemeris.next_bank_holiday(self)).downcase.to_sym
+        end
+
+        #
+        # Description of the closest holiday on or after this date.
+        #
+        # @param [String, nil] file Optional file to use for holiday definitions.
+        # @return [String]
+        #
+        def next_holiday_description(file: nil)
+          ::Ephemeris.get_holiday_description(next_holiday(file: file).to_s.upcase)
+        end
+
+        #
+        # Determines if this time is during a weekend.
+        #
+        # @return [true, false]
+        #
+        # @example
+        #   Time.now.weekend?
+        #
+        def weekend?
+          ::Ephemeris.weekend?(self)
+        end
+
+        #
+        # Determines if this time is during a specific dayset
+        #
+        # @param [String, Symbol] set
+        # @return [true, false]
+        #
+        # @example
+        #   Time.now.in_dayset?("school")
+        #
+        def in_dayset?(set)
+          ::Ephemeris.in_dayset?(set.to_s, self)
+        end
+
+        #
+        # Calculate the number of days until a specific holiday
+        #
+        # @param [String, Symbol, nil] holiday If not given, assumes the next holiday
+        # @param [String, nil] file Optional file to use for holiday definitions.
+        # @return [Integer]
+        # @raise [ArgumentError] if the holiday isn't valid
+        #
+        # @example
+        #   Time.now.days_until(holiday: :christmas) # => 2
```

Comment:
Wouldn't it be nicer to call `Time.now.days_until(:christmas)`?

Reply:
Done.